### PR TITLE
Add a --prefer-binary flag to 'pip install' command

### PR DESCRIPTION
**Pull request by tenzen-y**
Signed-off-by: Yuki Iwai <yuki.iwai.tz@gmail.com>

**What this PR does / why we need it**:
~~I pinned the h5py version with 3.7.0 to resolve image-building errors. These errors are caused by the latest h5py not releasing the wheel for aarch64.~~

I added a --prefer-binary flag to `pip install` command to resolve the following build errors.

Doc: https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-prefer-binary

Also, these errors are caused by the latest h5py not releasing the binary for aarch64.

```shell
...
 > [linux/arm64 stage-1 7/8] RUN pip install --no-cache-dir -r requirements.txt:
#27 136.0       running build_ext
#27 136.0       Building h5py requires pkg-config unless the HDF5 path is explicitly specified using the environment variable HDF5_DIR. For more information and details, see https://docs.h5py.org/en/stable/build.html#custom-installation
#27 136.0       error: pkg-config probably not installed: FileNotFoundError(2, 'No such file or directory')
#27 136.0       [end of output]
#27 136.0   
#27 136.0   note: This error originates from a subprocess, and is likely not a problem with pip.
#27 136.0 Successfully built googleapis-common-protos
#27 136.0   ERROR: Failed building wheel for h5py
#27 136.0 Failed to build h5py
#27 136.0 ERROR: Could not build wheels for h5py, which is required to install pyproject.toml-based projects
```

- https://github.com/kubeflow/katib/actions/runs/3995139030/jobs/6853668369
- https://github.com/kubeflow/katib/actions/runs/3995139035/jobs/6853668658
- https://github.com/kubeflow/katib/actions/runs/3995139039/jobs/6853668377
- https://github.com/kubeflow/katib/actions/runs/3995139039/jobs/6853671593

Ref: https://github.com/h5py/h5py/issues/2215

Blocking #2077 #2091 #2092 #2094

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes #

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

File: cmd/suggestion/hyperopt/v1beta1/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ ENV TARGET_DIR /opt/katib
 ENV SUGGESTION_DIR cmd/suggestion/hyperopt/v1beta1
 ENV PYTHONPATH ${TARGET_DIR}:${TARGET_DIR}/pkg/apis/manager/v1beta1/python:${TARGET_DIR}/pkg/apis/manager/health/python
 
-RUN if [ "${TARGETARCH}" = "ppc64le" ] || [ "${TARGETARCH}" = "aarch64" ]; then \
+RUN if [ "${TARGETARCH}" = "ppc64le" ]; then \
```
Comment (tenzen-y) [inline cmd/suggestion/hyperopt/v1beta1/Dockerfile:16]: BuildKit propagates `TARGETARCH` with `amd64` or `arm64`. However, we have been able to build container images for the platform linux/arm64 with an incorrect value, `aarch64`.

This means we don't need these packages for the platform `linux/arm64`.

**Comment by tenzen-y** [comment]
I found a better solution.
/hold

**Comment by johnugeorge** [comment]
LGTM
/assign @andreyvelich

**Review by andreyvelich — COMMENTED** [review]
Thank you for the prompt fix @tenzen-y !
/lgtm
/approve

**Pull request merged**